### PR TITLE
[TF Java] Adds support for adding control inputs via Ops API

### PR DESCRIPTION
**Pull request by irenedea**
Adds withControlDependencies to Ops API as discussed here: https://github.com/tensorflow/tensorflow/issues/26024

cc: @karllessard @sjamesr

### CODE REVIEW

**Comment by irenedea** [comment]
@melissagrueter @samdow

File: tensorflow/java/src/main/java/org/tensorflow/op/Scope.java [hunk 1]
```diff
@@ -146,11 +150,38 @@ public String makeOpName(String defaultName) {
     return nameScope.makeOpName(defaultName);
   }
 
-  private Scope(Graph graph, NameScope nameScope) {
+  /**
+   * Returns a new scope with where added operations will have the provided control dependencies.
```
[the remark this answers is not in the document]
Comment (irenedea) [reply tensorflow/java/src/main/java/org/tensorflow/op/Scope.java]: Good catch, thanks!

File: tensorflow/java/src/test/java/org/tensorflow/op/core/GeneratedOperationsTest.java [hunk 2]
```diff
@@ -57,4 +58,29 @@ public void testListInputTensorOutput() {
...
+        Session sess = new Session(g)) {
+      Ops ops = Ops.create(g);
+      Operand<Integer> variable = ops.variable(Shape.scalar(), Integer.class);
+      Operand<?> initVariable = ops.assign(variable, ops.constant(0));
+      ArrayList<Operand<?>> controls = new ArrayList<Operand<?>>();
+      controls.add(ops.assign(variable, ops.constant(3)));
+      Operand<Integer> x = ops.withControlDependencies(controls).math().add(ops.constant(0), ops.constant(0));
+      sess.runner().addTarget(initVariable).run();
+      try (Tensor<Integer> result = sess.runner().addTarget(x).fetch(variable).run().get(0).expect(Integer.class)) {
+        assertEquals(3, result.intValue());
```
[the remark this answers is not in the document]
Comment (irenedea) [reply tensorflow/java/src/test/java/org/tensorflow/op/core/GeneratedOperationsTest.java]: Wonderful suggestion :)

**Comment by irenedea** [comment]
friendly ping :) @mingxingtan

**Comment by mingxingtan** [comment]
Thanks for the PR.  It looks good to me, but I am not very familiar to specific Java APIs.  Will ask someone else to review it later. Thanks!

**Comment by irenedea** [comment]
@mingxingtan Thanks! Should we be adding someone as a reviewer to this PR?

cc @rthadur

**Comment by sjamesr** [comment]
Probably me (@sjamesr), I'll take a look soon

**Comment by irenedea** [comment]
Thanks @sjamesr! How can I add you as a reviewer for this PR?

**Comment by mingxingtan** [comment]
Just added @sjamesr as a reviewer. Thanks James!

**Comment by irenedea** [comment]
friendly ping :) @sjamesr

**Comment by karllessard** [comment]
Hey @irenedea ,  you might probably want to rebase your PR now to avoid any merge conflicts after approval, since [this huge one](https://github.com/tensorflow/tensorflow/pull/24858) has just been merged.

**Review by sjamesr — DISMISSED** [review]

**Comment by rthadur** [comment]
@irenedea please resolve conflicts

**Comment by irenedea** [comment]
@sjamesr @rthadur Rebased and resolved conflicts, please take another look, thanks!

**Comment by irenedea** [comment]
@sjamesr @rthadur friendly ping :)

**Review by sjamesr — APPROVED** [review]

**Comment by rthadur** [comment]
@irenedea can you please sign CLA

**Comment by irenedea** [comment]
Resolved CLA issues, should be signed now

**Comment by irenedea** [comment]
CLA issue fixed, should be signed now

**Comment by irenedea** [comment]
coauthors should have CLA now

**Comment by rthadur** [comment]
> CLA issue fixed, should be signed now

@irenedea it still shows CLA is pending , can you use please make sure to use same GitHub username and email-id associated with it.

**Comment by irenedea** [comment]
@rthadur CLA issues have been fixed, can you please help merge the PR? Thank you!

**Comment by irenedea** [comment]
@rthadur Should be fixed now, is it still pending?

**Comment by rthadur** [comment]
> @rthadur Should be fixed now, is it still pending?

Its resolved , starting internal merge, thank you

**Pull request merged**